projections.go

```diff
@@ -0,0 +1,125 @@
+package eventsourcing
+
+import (
+	"context"
+	"time"
+
+	"github.com/hallgren/eventsourcing/core"
+	"github.com/hallgren/eventsourcing/internal"
+)
+
+type Projection struct {
+	getF        func() (core.Iterator, error)
+	callbackF   func(e Event)
+	projections *Projections
+}
+
+type Projections struct {
+	register     *internal.Register // used to map the event types
+	deserializer DeserializeFunc
+	projections  []Projection
+}
+
+func NewProjections(register *internal.Register, deserializer DeserializeFunc) *Projections {
+	return &Projections{
+		register:     register,
+		deserializer: deserializer,
+		projections:  make([]Projection, 0),
+	}
+}
+
+func (p *Projections) Add(getF func() (core.Iterator, error), callbackF func(e Event)) *Projection {
+	projection := Projection{
+		getF:        getF,
+		callbackF:   callbackF,
+		projections: p,
+	}
+	p.projections = append(p.projections, projection)
+	return &projection
+}
+
+// Start starts all projections
+func (p *Projections) Start() {
+
+}
+
+// Close closes all projections
+func (p *Projections) Close() {
+	for _, projection := range p.projections {
+		projection.Close()
+	}
+}
+
+// Run runs the projection forever until the context is cancelled
+// When there is no more events to concume it sleeps the pace and run again.
```

Comment:
```suggestion
// Run runs the projection forever until the context is cancelled.
// When there are no more events to consume it sleeps the pace and run again.
```

Reply:
I close this as it has passed alot of water since the review and it was conducted when the PR was in a early stage.